eth/stagedsync/stage_bor_heimdall.go

```diff
@@ -206,12 +202,14 @@ func BorHeimdallForward(
 		nextSpanId = binary.BigEndian.Uint64(k) + 1
 	}
 	snapshotLastSpanId := cfg.blockReader.(LastFrozen).LastFrozenSpanID()
-	if snapshotLastSpanId+1 > nextSpanId {
+	if s.BlockNumber != 0 && snapshotLastSpanId+1 > nextSpanId {
```

Comment:
This is needed because even though we don't have 0th span, the function `LastFrozenSpanID` will return 0 which makes this check assume that we have 0th span and need the 1st span. This is not the case when we start to mine from scratch in devnets.